aks-kubesphere-linux/azuredeploy.json

```diff
@@ -0,0 +1,506 @@
+            "name": "[parameters('virtualNetworkName')]",
+            "type": "Microsoft.Network/VirtualNetworks",
+            "apiVersion": "2019-04-01",
+            "location": "[variables('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('NetworkSecurityGroupName'))]"
+            ],
+            "tags": {},
+            "properties": {
+                "addressSpace": {
+                    "addressPrefixes": "[parameters('addressSpaces')]"
+                },
+                "subnets": [
+                    {
+                        "name": "[parameters('AKSSubnetName')]",
+                        "properties": {
+                            "addressPrefix" : "[parameters('AKSSubnetAddressRange')]",
+                            "networkSecurityGroup": {
+								"id": "[variables('networkSecurityGroupID')]"
+							}
+                        }
+                    },
+                    {
+                        "name": "[parameters('ClientSubnetName')]",
+                        "properties": {
+                            "addressPrefix": "[parameters('ClientSubnetAddressRange')]",
+                            "networkSecurityGroup": {
+								"id": "[variables('networkSecurityGroupID')]"
+							}
+                        }
+                    }
+                ],
+                "enableDdosProtection": "[variables('ddosProtectionPlanEnabled')]"
+            }
+        },
+		{
+			"apiVersion": "2019-09-01",
+			"type": "Microsoft.Network/networkSecurityGroups",
+			"name": "[parameters('NetworkSecurityGroupName')]",
+			"location": "[variables('location')]",
+			"tags": {},
+			"properties": {
+				"securityRules": [
+					{
+						"name": "allowSSHin_all",
+						"properties": {
+							"description": "Allow SSH in from all locations",
+							"protocol": "Tcp",
+							"sourcePortRange": "*",
+							"destinationPortRange": "22",
+							"sourceAddressPrefix": "*",
+							"destinationAddressPrefix": "*",
+							"access": "Allow",
+							"priority": 100,
+							"direction": "Inbound"
+						}
+					},
+					{
+						"name": "allowKSConsolein_all",
+						"properties": {
+							"description": "Allow KS Console in from all locations",
+							"protocol": "Tcp",
+							"sourcePortRange": "*",
+							"destinationPortRange": "30880",
+							"sourceAddressPrefix": "*",
+							"destinationAddressPrefix": "*",
+							"access": "Allow",
+							"priority": 210,
+							"direction": "Inbound"
+						}
+					}
+				]
+			}
+		},
+		{
+			"name": "AKSDeployment",
+			"type": "Microsoft.Resources/deployments",
+			"apiVersion": "2019-05-01",
+			"dependsOn": [
+				"[resourceId('Microsoft.Network/virtualNetworks/', parameters('virtualNetworkName'))]"
+			],
+			"properties": {
+				"mode": "Incremental",
+				"templateLink": {
+					"uri": "[variables('AKSDeploymentTemplateUrl')]",
+					"contentVersion": "1.0.0.0"
+				},
+				"parameters": {
```

Comment:
format the json, e.g. SHIFT+ALT+F in VS Code

Reply:
Oh my god, the mixed-use of tabs and spaces caused this problem, I'm so sorry.